### DISCUSSION TITLE
Enabling `keepalive` if a physical action exists

### DISCUSSION THREAD

**Discussion by Soroosh129** [lf-lang/lingua-franca#547] [General]
Imagine the following LF program:
```python
target Python;
reactor Foo {
    physical action a;
    reaction(a) {=
        print("A physical event has occurred")
    =}
}
main reactor {
    f = new Foo();
}
```
This program exits as soon as it starts. The only reason for this is that the (at least C and Python) runtimes think that no events exist in this program and `keepalive` is set to false. Therefore, they decide to exit the program. However, if a physical action exists, there is no way for the runtime to make such a determination by itself, since a physical action can produce an event at any given moment in perpetuity.

This issue was recently rediscovered by @housengw while trying to implementation the C target sensor simulator in the Python target.

I would suggest that to err on the safe side, we should set `keepalive` to true when physical actions exist in a program, unless the user has explicitly set it to false. What do you think?

I have created a draft PR that reflects the changes required for this feature in #546.

**Comment by edwardalee** [comment]
I thought we already had this.  In any case, I think we should definitely set keepalive to true if there are physical actions.

**Comment by lhstrh** [comment]
https://github.com/lf-lang/lingua-franca/pull/546 has been merged into master.
